assets/src/scripts/components/dailyValueHydrograph/selectors/legend-data.js

```diff
@@ -0,0 +1,74 @@
+import {createSelector} from 'reselect';
+import {getCurrentTimeSeriesLineSegments} from './time-series-data';
+import {defineLineMarker, defineTextOnlyMarker} from '../../../d3-rendering/markers';
+
+
+const tsLineMarkers = function(lineClasses) {
+    let result = [];
+
+    if (lineClasses.default) {
+        result.push(defineLineMarker(null, 'line-segment ts-dv', 'Provisional'));
+    }
+    if (lineClasses.approved) {
+        result.push(defineLineMarker(null, 'line-segment approved ts-dv', 'Approved'));
+    }
+    if (lineClasses.estimated) {
+        result.push(defineLineMarker(null, 'line-segment estimated ts-dv', 'Estimated'));
+    }
+    return result;
+};
+
+
+/**
+ * create elements for the legend in the svg
+ *
+ * @param {Object} displayItems - Object containing Object containing default, estimated and approved properties.
+ * @return {Array} - Returns an array of markers.
+ */
+const createLegendMarkers = function(displayItems) {
+    const legendMarkers = [];
+
+    if (displayItems) {
+        const currentMarkers = [
+            ...tsLineMarkers(displayItems)
+        ];
+        if (currentMarkers.length) {
+            legendMarkers.push([
+                defineTextOnlyMarker('', null, 'ts-legend-current-text'),
```

Comment:
This is where the extra marker is coming in. We can remove this line because it really doesn't render any thing. Then the tests will only have 2 line markers. Regarding the other issue about removing the showSeries: {current: true}. I removed that and did not have an issue. I will email an updated legend-data.spec.js that has this change and the change to the tests for only two markers.

Reply:
Sorry I didn't catch this before, this is fixed now.